routes/companies.js

```diff
@@ -0,0 +1,152 @@
+var express = require('express');
+var router = express.Router();
+var Company = require('../models/company')
+
+router.get('/', function (request, response, next) {
+  index(request, response)
+});
+
+router.get('/new', function (request, response, next) {
+  response.render('companies/new', { title: 'New Company', layout: './layouts/application' });
+});
+
+router.post('/', (request, response) => {
+  create(request, response)
+})
+
+router.get('/:id/edit', (request, response) => {
+  edit(request, response)
+})
+
+router.get('/:id', (request, response) => {
+  show(request, response)
+})
+
+router.post('/:id', (request, response) => {
+  update(request, response)
+})
+
+router.get('/:id/delete', (request, response) => {
+  destroy(request, response)
+})
+
+async function index(request, response) {
+  let db = request.app.get('db')
+  let results = await db.search(
+    'companies',
+    {} // TODO blank check in db.search or?
```

Comment:
wouldn't it be nice if you had a `Model.all` method?

Reply:
and a `where` too